src/game/AI/ScriptDevAI/scripts/world/spell_scripts_wotlk.cpp

```diff
@@ -99,11 +99,63 @@ struct BloodReserveEnchant : public AuraScript
     }
 };
 
+// 64415 Val'anyr Hammer of Ancient Kings - Equip Effect
+struct ValanyrEquipEffect : public AuraScript
+{
+    bool OnCheckProc(Aura* /*aura*/, ProcExecutionData& procData) const override
+    {
+        if (procData.healthGain == 0)
+            return false;
+        return true;
+    }
+
+    SpellAuraProcResult OnProc(Aura* aura, ProcExecutionData& procData) const override
+    {
+        procData.cooldown = 45;
+        return SPELL_AURA_PROC_OK;
+    }
+};
+
+// 64411 Blessing of Ancient Kings
+struct BlessingOfAncientKings : public AuraScript
+{
+    SpellAuraProcResult OnProc(Aura* aura, ProcExecutionData& procData) const override
+    {
+        if (!IsPositiveSpell(aura->GetId()))
```

Comment:
snagged that from the original UnitAuraProcHandler.cpp and I'm not sure what it does, so I kept it. purely from the name it doesn't make much sense, tho